api/agent/drivers/docker/docker.go

```diff
@@ -39,7 +38,7 @@ type Auther interface {
 	// certain restrictions on images or if credentials must be acquired right
 	// before runtime and there's an error doing so. If these credentials don't
 	// work, the docker pull will fail and the task will be set to error status.
-	DockerAuth() (*docker.AuthConfiguration, error)
+	DockerAuth(ctx context.Context) (*docker.AuthConfiguration, error)
```

Comment:
This is a breaking change as it is changing an existing Interface am I right? The right process would be to create a new interface and have a kind of deprecation path? I know it is a pain I am just interested in your opinion and understand if we have a process to deal with these kind of changes.

Reply:
i was not able to find any third party use of this package and all other required changes lie here. thanks for catching this - I should've made a note.